# Acral melanoma exomes' analysis
# Copy number alterations
## Author: Patricia Basurto Lozada

## Copy number calling

Estimation of copy number, ploidy and cellularity was done using SEQUENZA. Fewer samples were used for copy number alteration analysis as some sample were filtered out if the had very low cellularity estimated by sequenza (lower than 20% tumor cellularity) or data was observed with oversegmentation in sequenza plots. 

To identify significant regions altered by copy number alteration we used the toool GISTIC2, using as input one sample per patient (giving preference to primary tumors when available).

## Identifying significant peaks with GISTIC2 

We generated GISTIC2 input file using sequenza segmentation data outputs and formatting as istructed in GISTIC2 documentation. 

GISTIC2 was used with default parameters. 

``` bash
#!/bin/sh
## Setting up trhe GISTIC2 script
unset DISPLAY
## output directory
echo --- creating output directory ---
basedir=`pwd`/paper_acrals_hg38_one_per_patient
mkdir -p $basedir 

echo --- running GISTIC ---
## input file definitions
segfile=`pwd`/Paper_one_per_patient_cnlog_segments_for_gistic.txt
refgenefile=`pwd`/refgenefiles/hg38.UCSC.add_miR.160920.refgene.mat

## call script that sets MCR environment and calls GISTIC executable 
./gistic2 -b $basedir -seg $segfile  -refgene $refgenefile -genegistic 1 -savegene 1 

##-alf $alf 

##-cnv $cnvfile -genegistic 1 -smallmem 1 -broad 1 -brlen 0.5 -conf 0.90 -armpeel 1 -savegene 1 -gcm extreme

```
``` bash
#Running the previous script 
./run_gistic_acrals_per_patient_filtered
```

## Plotting GISTIC2 peaks with maftools

We plotted the significant regions affected by amplifications and deletions using the R package maftools, and using as input the outputs generated by GISTIC2.  


``` R
library(maftools)

all.lesions = ("all_lesions.conf_75.txt")
amp.genes = ("amp_genes.conf_75.txt")
del.genes = ("del_genes.conf_75.txt")
scores.gis = ("scores.gistic") 

gistic_data = readGistic( gisticAllLesionsFile = all.lesions, gisticAmpGenesFile = amp.genes, gisticDelGenesFile = del.genes, gisticScoresFile = scores.gis, cnLevel = "all", isTCGA= FALSE, verbose = TRUE)
gisticChromPlot(gistic= gistic_data, ref.build ="hg38",  markBands= FALSE)

```

## Generating CN Heatmap

Data for the heatmap was obtained from the web application CNApp using as input sequenza segmentation and copy number estimations, and using default values. Values by chromosome arms were used. Samples were ordered based on mutational status. 

``` python
#Reading file ordered by mutational status
cn_data = pd.read_csv("ordered_cn_heatmap.csv", sep="\t", index_col=0)
sns.set(rc={'figure.figsize':(30,15)})
#Plotting hetamap
sns.heatmap(cn_data, center=0, cmap="bwr", robust=True)
plt.savefig("CN_by_mut_heatmap.pdf", format='pdf')
```

# References 

SEQUENZA
Favero, F et al. “Sequenza: allele-specific copy number and mutation profiles from tumor sequencing data.” Annals of oncology : official journal of the European Society for Medical Oncology vol. 26,1 (2015): 64-70. doi:10.1093/annonc/mdu479

CNApp
Franch-Expósito, Sebastià et al. “CNApp, a tool for the quantification of copy number alterations and integrative analysis revealing clinical implications.” eLife vol. 9 e50267. 15 Jan. 2020, doi:10.7554/eLife.50267
github: https://github.com/ait5/CNApp

maftools
Mayakonda A, Lin DC, Assenov Y, Plass C, Koeffler HP. 2018. Maftools: efficient and comprehensive analysis of somatic variants in cancer. Genome Research. PMID: 30341162
github: https://github.com/PoisonAlien/maftools